rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -128,6 +128,12 @@ class PublisherBase
   size_t
   get_intra_process_subscription_count() const;
 
+  /// Get actual qos settings
+  /** \return The number of subscriptions. */
+  RCLCPP_PUBLIC
+  std::shared_ptr<rmw_qos_profile_t>
```

Comment:
Should this just be `rmw_qos_profile_t` since it's trivially copied, why use a shared pointer?

Reply:
I have two other options:
```cpp
bool get_actual_qos(rmw_qos_profile_t & qos_out)
rmw_qos_profile_t get_actual_qos()
```
I discarded the second one, because it's not way to identify an error.
If you prefer, I can modify this to use the first signature.

Reply:
> I discarded the second one, because it's not way to identify an error.

Why not throw an exception?